tasks/flush_repo.py

```diff
@@ -151,10 +155,32 @@ def _delete_label_analysis(self, db_session: Session, commit_ids, repoid: int):
 
     @sentry_sdk.trace
     def _delete_commits(self, db_session: Session, repoid: int) -> int:
-        deleted_commits = db_session.query(Commit).filter_by(repoid=repoid).delete()
+        commits_to_delete = (
+            db_session.query(Commit.commitid).filter_by(repoid=repoid).all()
+        )
+        commit_ids_to_delete = [commit.commitid for commit in commits_to_delete]
+
+        pipeline = get_redis_connection().pipeline()
+        for id in commit_ids_to_delete:
+            pipeline.set(
+                get_parallel_upload_processing_session_counter_redis_key(
+                    repoid=repoid, commitid=id
+                ),
+                0,
+            )
```

Comment:
i'm not positive but i think `set()` will undo any expiry that was set on the key. i think this method takes a keyword argument `ex` where you can pass in the expiry https://redis-py.readthedocs.io/en/stable/commands.html#redis.commands.core.CoreCommands.set

Reply:
Hmm... Are you saying this because it is not sufficient to set a key explicitly to 0? because it will still take up space? 

So what I need to do is set the key to 0, but still set a ttl aswell?